cmd/internal/migration/vpa.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package migration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+)
+
+// MigrateEmptyVPAPatch performs an empty patch updates to VerticalPodAutoscaler resources,
+// that are elighible to adopt the InPlaceOrRecreate update mode, but are filtered out, because of
+// the Resource Manager's alwaysUpdate=false configuration.
+// TODO(vitanovs): Remove the migration once the VPAInPlaceUpdates feature gates promoted to GA.
+func MigrateEmptyVPAPatch(ctx context.Context, mgr manager.Manager, log logr.Logger) error {
+	log.Info("Migrating VerticalPodAutoscalers")
+
+	vpaList := vpaautoscalingv1.VerticalPodAutoscalerList{}
+	if err := mgr.GetClient().List(ctx, &vpaList); err != nil {
+		if meta.IsNoMatchError(err) {
+			log.Info("Resources kind not found, skipping migration", "kind", "VerticalPodAutoscaler")
+			return nil
+		}
+		return fmt.Errorf("failed listing VerticalPodAutoscaler resources: %w", err)
+	}
+
+	for _, vpa := range vpaList.Items {
+		vpaHasSkipLabel := metav1.HasLabel(vpa.ObjectMeta, resourcesv1alpha1.VPAInPlaceUpdatesSkip)
+		vpaHasMutatedLabel := metav1.HasLabel(vpa.ObjectMeta, resourcesv1alpha1.VPAInPlaceUpdatesMutated)
+		if vpaHasSkipLabel || vpaHasMutatedLabel {
+			continue
+		}
+
+		if vpa.Namespace == metav1.NamespaceSystem || vpa.Namespace == v1beta1constants.KubernetesDashboardNamespace {
+			continue
+		}
+
+		vpaKey := client.ObjectKeyFromObject(&vpa)
+		log.Info("Updating VerticalPodAutoscaler resource", "vpa", vpaKey)
+
+		patch := client.MergeFrom(vpa.DeepCopy())
+		if err := mgr.GetClient().Patch(ctx, &vpa, patch); err != nil {
```

Comment:
```suggestion
		if err := mgr.GetClient().Patch(ctx, &vpa, client.RawPatch(types.MergePatchType, []byte("{}")); err != nil {
```

You could send an empty patch directly.

Reply:
Nice suggestion :)) Decided to switch to it, thanks!